firmware_mod/www/cgi-bin/ui_system.cgi

```diff
@@ -78,6 +81,16 @@ if [ -n "$F_cmd" ]; then
 		echo "<p>Setting wifi password to: $wifi_password</p>"
 		wpa_config_set psk "\"$wifi_password\""
 	fi
+  if [ -n ${F_ssh_port} ]; then
+		ssh_port=$(printf '%b' "${F_ssh_port//%/\\x5}")
+		echo "<p>Changing SSH port to: $ssh_port</p>"
+		rewrite_config /system/sdcard/config/ssh.conf ssh_port "$ssh_port"
+	fi
+  if [ -n ${F_ssh_key} ]; then
+		ssh_key=$(printf '%b' "${F_ssh_key//%/\\x}")
```

Comment:
The spaces in the key get swapped for +'s for example "my+test+key"

Reply:
Plus signs in CGI parameters mean spaces. Apparently that's in the CGI spec.
Maybe run javascript to encode the spaces to %20 before sending to the key to the cgi script.
Then in the shell script use regex to replace to decode the spaces to real spaces again.

Reply:
Thanks, that worked perfectly.  Will submit a final version of this patch tomorrow